### DISCUSSION TITLE
1-way push from Obsidian to CalDAV source

### DISCUSSION THREAD

**Discussion by davish** [davish/obsidian-full-calendar#346] [Feature Requests]
From #1, @dbarenholz writes:
> Any event you create in obsi plugin is pushed to a calendar in gcal, but changes in gcal aren't propagated back.

**Comment by Gr-Nobel** [comment]
Hey i can't create event in obsidian that syncs to the google calendar. I have copied the private ics links and pasted them . I can only view the event but can't create them in obsidian to google calendar

**Comment by davish** [comment]
@Gr-Nobel This feature is not implemented yet, which is why this issue is still open.

**Comment by Gr-Nobel** [comment]
The plugin got an update is this feature  implemented

**Comment by davish** [comment]
No. I will close this issue if this feature is implemented.

**Comment by vigoux** [comment]
Hey, sorry to be pushy, but do you have any idea of the schedule for that issue to get fixed ?
If needed I am willing to contribute a fix, but I can't guarantee neither that it'll fast, nor well-written...

**Comment by vigoux** [comment]
Looking at the code a little, it seems that the changes are somewhat non-trivial:
- The `EditableCalendar` type is really only a shim above `NoteCalendar` the actual work to make it more generic is actually somewhat big
- The `EditableCalendar` API itself is tied to _file paths_ instead of _event UIDs_.

I am going to attempt to have a look into this and try to open a pull request, but I can't guarantee anything.

**Comment by davish** [comment]
Hi @vigoux, as you observed, the abstractions like `EditableCalendar` in the `models` directory are really not very good. To address the problems you've brought up, I'm in the middle of completely rewriting most of the core plugin logic. The first step of that was writing a new cache layer that got merged in #204. I'm working on wiring that up to the four types of calendars currently supported (full notes, daily notes, readonly ICS, and readonly CalDAV) in a local branch that I have not yet pushed up.

I would advise against starting any significant pull request based off of the current master branch that involves the core plugin logic, since the idea is to delete the `models` folder and rely on other, better separated, classes and abstractions. I won't be accepting PRs that add functionality to the old framework. I will let you know back on this issue when I've posted a PR for the second half of the rewrite.

**Comment by vigoux** [comment]
Hey, thanks for the heads up.
If needed I will gladly test that PR you are mentionning, as this would be a great enabler on my side.

Thanks again for your time.
